modules/core/arrow-core/src/main/kotlin/arrow/core/extensions/sortedmap.kt

```diff
@@ -111,3 +118,36 @@ interface SortedMapKHash<K : Comparable<K>, A> : Hash<SortedMapK<K, A>>, SortedM
       31 * hash + HA().run { a.hash() }
     }
 }
+
+interface SortedMapKSemialign<K : Comparable<K>> : Semialign<SortedMapKPartialOf<K>>, SortedMapKFunctor<K> {
+  override fun <A, B> align(
+    a: Kind<SortedMapKPartialOf<K>, A>,
+    b: Kind<SortedMapKPartialOf<K>, B>
+  ): Kind<SortedMapKPartialOf<K>, Ior<A, B>> {
+    val l = a.fix()
+    val r = b.fix()
+    val keys = l.keys + r.keys
+
+    return keys.map { key ->
+      Ior.fromOptions(l[key].toOption(), r[key].toOption()).map { key to it }
+    }.flattenOption().toMap().toSortedMap().k()
+  }
+}
+
+fun <K : Comparable<K>> SortedMapK.Companion.semialign(): SortedMapKSemialign<K> =
+  object : SortedMapKSemialign<K> {}
+
+interface SortedMapKAlign<K : Comparable<K>> : Align<SortedMapKPartialOf<K>>, SortedMapKSemialign<K> {
+  override fun <A> empty(): Kind<SortedMapKPartialOf<K>, A> = emptyMap<K, A>().toSortedMap().k()
+}
+
+fun <K : Comparable<K>> SortedMapK.Companion.align(): SortedMapKAlign<K> =
+  object : SortedMapKAlign<K> {}
+
+@extension
```

Comment:
Wait, does this work? If so, what is the reason all the others are not using `@extension`? Can you test if this generates the right code and then maybe change the other instances as well?

Reply:
adding `@extension` to SortedMapKSemialign leads to

```
internal val semialign_singleton: SortedMapKSemialign<Any?> = object :
    SortedMapKSemialign<java.lang.Comparable> {}

Type argument is not within its bounds: should be subtype of 'Comparable<Any?>'
One type argument expected for interface Comparable<T : Any!>
```

similiar effect with the Align instance

Reply:
Ah ok, hopefully those things will be gone with arrow-meta in the future...